Library/Homebrew/test/caveats_spec.rb

```diff
@@ -26,4 +26,36 @@ def caveats
       expect(described_class.new(f)).not_to be_empty
     end
   end
+
+  describe "#caveats" do
+    context "when f.plist is not null" do
+      it "prints plist startup information when plist_startup is not null" do
+        allow(f).to receive(:plist).and_return(true)
+        allow(f).to receive(:plist_startup).and_return(true)
+        expect(subject.caveats).to include("To have launchd start #{f.full_name} now and restart at startup:\n  sudo brew services start #{f.full_name}\n")
+      end
+
+      it "prints plist login information when f.plist_startup is null" do
```

Comment:
Two small nits: Above you are using only `plist_startup`, here you are using `f.plist_startup`. Secondly, use `nil` instead of `null`, to be consistent with Ruby.